rcsb/db/mongo/PdbxLoader.py

```diff
@@ -783,8 +787,18 @@ def loadWorker(self, dataList, procName, optionsD, workingDir):
                     cId = cL[0].getName() if useNameFlag else cL[0].getProp("uid")
                     cIdD[cId] = locatorObj
                     containerList.extend(cL)
-            # -- Apply methods to each container -
+            #
             for container in containerList:
+                # -- Assign temporary attribute with internal identifier for computed models
+                if databaseName in ["pdbx_comp_model_core"]:
+                    eObj = container.getObj("entry")
+                    sourceId = eObj.getValue("id", 0)
+                    compModelInternalId = compModelIdMapD.get(sourceId, sourceId)
+                    if sourceId == compModelInternalId:
+                        logger.warning("Unable to map computed-model sourceId (%s) to internal identifier. Will use original source ID as internal ID (rcsb_id).", sourceId)
+                    eObj.appendAttribute("rcsb_comp_model_id")
```

Comment:
Perhaps better to check if the attribute exists before appending.

Reply:
OK, I can add a check and if it is already present, log an error. In normal flow, that attribute shouldn't already exist at this stage (the initial file reading stage), but if it does for whatever reason I agree it'd be good to catch that.